src/middleware/authentication.ts

```diff
@@ -0,0 +1,22 @@
+import { AuthenticationError } from "apollo-server";
+import { logger } from "../logger";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+export const authenticationContextMiddleware = ({ req }: { req: any }) => {
+  const authHeaderUser: string = req.headers.authorization || "";
+  const expectedHeader: string = process.env.AUTH_TOKEN!;
```

Comment:
To avoid using ! here, it's better to load these environment variables in a separate file, such as environment.ts. In this file, you can create an env = {} object and a utility function that retrieves the environment variables. This function should throw an exception if a required value is missing. By handling potential errors at the beginning, you ensure that any issues with missing environment variables are caught early, preventing runtime problems later.

This is something a little bit more complex than what you [need](https://github.com/t3-oss/t3-env) but you can get inspired by it.

Reply:
I created an environment file where I check the environment variables. I made the use of the singleton design pattern as there should be only one environment and instanciated the variables in an object which I export.